### ISSUE TITLE
How to use autogluon to rank?

### ISSUE THREAD

**Issue by Gpwner** [awslabs/autogluon#2264]
I've been looking for an auto-ml framework that solves learning to rank  problem, but I haven't found one.Does autogluon support?Thanks.

**Comment by Innixma** [comment]
AutoGluon currently doesn't support the ranking task explicitly, but we are considering to add this in future.

For now, you could treat the problem as regression / binary classification and calculate the ranks based on the prediction probabilities. This isn't a perfect solution, but could work.

For example, https://towardsdatascience.com/learning-to-rank-a-complete-guide-to-ranking-using-machine-learning-4c9688d370d4

You can treat the prediction probability as the score `s` when ranking, thus using AutoGluon as a `pointwise ranker` or `pairwise ranker`. The suboptimal part of this is you wouldn't be able to specify a special ranking loss or evaluation metric (aka doesn't support `listwise ranker`), but if AutoGluon's model is generally more powerful than the model you alternatively use that does have dedicated ranking support, AutoGluon might still be the best choice for the task.

**Comment by Gpwner** [comment]
That is great!

**Comment by Innixma** [comment]
@Gpwner Glad my answer was useful to you! If you happen to try out AutoGluon for ranking in the manner I described, I'd be interested to hear how it performs!

**Comment by samvelkoch** [comment]
Any news on the topic ? @Innixma

**Comment by samvelkoch** [comment]
@Innixma [CatBoost](https://catboost.ai/docs/en/concepts/python-reference_catboostranker), [LightGBM](https://lightgbm.readthedocs.io/en/latest/pythonapi/lightgbm.LGBMRanker.html) and [XGBoost](https://xgboost.readthedocs.io/en/latest/tutorials/learning_to_rank.html)t have ranking onboard btw. Flaml integrated last 2 but not that perfect implementation imo (the have no OOF for example in class)

**Comment by Innixma** [comment]
@samvelkoch Maybe in the future, we currently have a lot of high priority tasks that are more urgent, as supporting a new problem type is time consuming and requires a new benchmark. If another AutoML system has ranking support with ensembles, that would be something I might take a look at.

**Comment by samvelkoch** [comment]
@Innixma thanks fir answering Nick. Yes, [FLAML](https://microsoft.github.io/FLAML/docs/Examples/AutoML-Rank#a-simple-learning-to-rank-example) has it on board.
